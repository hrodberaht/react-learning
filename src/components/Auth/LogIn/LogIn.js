import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class LogIn extends Component {
    state = {
      login: '',
      email: '',
    }

    handleChange = (e) => {
      this.setState({ [e.target.id]: e.target.value });
    }

    handleSubmit = (e) => {
      e.preventDefault();

      this.setState({
        login: '',
        email: '',
      });
    }

    render() {
      const { login, email } = this.state;
      return (
        <form onSubmit={this.handleSubmit}>
          <input id="login" value={login} onChange={this.handleChange} />
          <input id="email" value={email} onChange={this.handleChange} />
          <button type="submit">Log In</button>
        </form>
      );
    }
}

LogIn.propTypes = {

};
