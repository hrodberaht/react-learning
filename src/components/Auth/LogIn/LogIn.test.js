import React from 'react';
import { shallow } from 'enzyme';
import { LogIn } from './LogIn';

describe('<LogIn />', () => {
  let wrapper;
  const props = {
    handleLogIn: jest.fn(),
    history: [],
  };

  const setValuesToInputs = () => {
    wrapper.find('#login').simulate('change', { target: { value: 'John', id: 'login' } });
    wrapper.find('#password').simulate('change', { target: { value: 'test@test.pl', id: 'password' } });
  };
  beforeEach(() => {
    wrapper = shallow(<LogIn {...props} />);
  });

  it('should component render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call function handleChange', () => {
    wrapper.instance().handleChange = jest.fn();
    wrapper.update();
    setValuesToInputs();
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  });

  it('should change state after change input', () => {
    setValuesToInputs();
    expect(wrapper.state()).toEqual({ login: 'John', password: 'test@test.pl' });
  });

  it('should call handleLogIn function with login and password', () => {
    setValuesToInputs();
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(props.handleLogIn).toHaveBeenCalled();

    expect(wrapper.state()).toEqual({ login: '', password: '' });
  });
});
