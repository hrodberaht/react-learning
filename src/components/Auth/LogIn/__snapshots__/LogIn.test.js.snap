// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> should component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn
    handleLogIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="login"
>
          Login:
          <input
                    id="login"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
</label>,
        <label
          htmlFor="email"
>
          Email:
          <input
                    id="email"
                    onChange={[Function]}
                    type="email"
                    value=""
          />
</label>,
        <button
          type="submit"
>
          Log In
</button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Login:",
            <input
              id="login"
              onChange={[Function]}
              type="text"
              value=""
/>,
          ],
          "htmlFor": "login",
        },
        "ref": null,
        "rendered": Array [
          "Login:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "login",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Email:",
            <input
              id="email"
              onChange={[Function]}
              type="email"
              value=""
/>,
          ],
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": Array [
          "Email:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "onChange": [Function],
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log In",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Log In",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="login"
>
            Login:
            <input
                        id="login"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
</label>,
          <label
            htmlFor="email"
>
            Email:
            <input
                        id="email"
                        onChange={[Function]}
                        type="email"
                        value=""
            />
</label>,
          <button
            type="submit"
>
            Log In
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Login:",
              <input
                id="login"
                onChange={[Function]}
                type="text"
                value=""
/>,
            ],
            "htmlFor": "login",
          },
          "ref": null,
          "rendered": Array [
            "Login:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "login",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Email:",
              <input
                id="email"
                onChange={[Function]}
                type="email"
                value=""
/>,
            ],
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": Array [
            "Email:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
